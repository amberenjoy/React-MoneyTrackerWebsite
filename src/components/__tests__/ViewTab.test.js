import React from 'react';
import { shallow } from 'enzyme';
import ViewTab from '../ViewTab';
import { LIST_VIEW, CHART_VIEW } from '../../utility';

const props = {
    activeTab: LIST_VIEW,
    onTabChange: jest.fn(),
};
let wrapper;

describe('test ViewTab component ', () => {
    beforeEach(() => {
        wrapper = shallow(<ViewTab {...props} />)
    });
    it('Should render the component to match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.nav-item').length).toEqual(2);
    });
    it('should trigger the correct function callback', () => {
        const firstItem = wrapper.find('.nav-item').first();
        firstItem.find('a').first().simulate('click', {
            preventDefault: () => { }
        });
        expect(props.onTabChange).toHaveBeenCalled();
    });
})