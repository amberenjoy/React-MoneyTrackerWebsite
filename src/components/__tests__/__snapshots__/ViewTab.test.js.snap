// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ViewTab component  Should render the component to match the snapshot 1`] = `ShallowWrapper {}`;

exports[`test ViewTab component  ViewTab component renders correctly 1`] = `ShallowWrapper {}`;
